### ISSUE TITLE
Update wiki on particle attributes

### ISSUE THREAD

**Issue by n01r** [ComputationalRadiationPhysics/picongpu#1480]
Since the introduction of the [OpenPMD](https://github.com/openPMD) standard some changes have been made to how new attributes for particle species are defined. Specifically in `speciesAttributes.unitless`.   
The necessary steps should be documented in the [wiki](https://github.com/ComputationalRadiationPhysics/picongpu/wiki) because the error messages on forgetting to set `Unit<>`, `UnitDimension<>`, `MacroWeighted<>` and `WeightingPower<>` are quite nasty :snail: :interrobang:  and hard to trace back to their origin. 
Ideally there should be also a link to the [OpenPMD tutorial](http://www.openpmd.org/#/start) for these objects.

**Comment by ax3l** [comment]
Actually those are two, actually three things.

### PIConGPU
- input files, issue #12 
- `unitless` files are actually developer files, not user files (for wiki)

### HDF5 / ADIOS Output
- [own plugin page](https://github.com/ComputationalRadiationPhysics/picongpu/wiki/Plugin%3A-HDF5), link to file format updated

**Comment by ax3l** [comment]
To give you a quick intro about the first two points (someone might want to write those into #12), see the links to the in-code documentation:

- each particle attribute defined in [`speciesAttributes.param`](https://github.com/ComputationalRadiationPhysics/picongpu/blob/71cfb03ec813c88a7e6f304b870f8731093b3b4f/src/picongpu/include/simulation_defines/param/speciesAttributes.param#L37-L41) needs to be further described in [`speciesAttributes.unitless`](https://github.com/ComputationalRadiationPhysics/picongpu/blob/71cfb03ec813c88a7e6f304b870f8731093b3b4f/src/picongpu/include/simulation_defines/unitless/speciesAttributes.unitless#L39-L87)
- required traits for each attribute:
  - [`Unit`](https://github.com/ComputationalRadiationPhysics/picongpu/blob/71cfb03ec813c88a7e6f304b870f8731093b3b4f/src/picongpu/include/traits/Unit.hpp#L30-L40)
```
    /** Get unit of a date that is represented by an identifier
     *
     * \tparam T_Identifier any PIConGPU identifier
     * \return \p std::vector<float_64> ::get() as static public method
     *
     * Unitless identifies, see \UnitDimension, can still be scaled by a
     * factor. If they are not scaled, implement the unit as 1.0;
     * \see simulation_defines/unitless/speciesAttributes.unitless
     */
```
  - [`UnitDimension`](https://github.com/ComputationalRadiationPhysics/picongpu/blob/71cfb03ec813c88a7e6f304b870f8731093b3b4f/src/picongpu/include/traits/UnitDimension.hpp#L30-L45)
```
    /** Get power of seven SI base units of date that is represented by an identifier
     *
     * Definition must follow the openPMD `unitDimension` definition:
     * length L, mass M, time T, electric current I, thermodynamic temperature
     * theta, amount of substance N, luminous intensity J
     *   \see http://www.openPMD.org
     *   \see http://dx.doi.org/10.5281/zenodo.33624
     * Must return a vector of size() == 7, for unitless attributes all
     * elements are zero.
     *
     * \tparam T_Identifier any picongpu identifier
     * \return \p std::vector<float_64> ::get() as static public method
     *
     */
```
  - [`WeightingPower`](https://github.com/ComputationalRadiationPhysics/picongpu/blob/71cfb03ec813c88a7e6f304b870f8731093b3b4f/src/picongpu/include/particles/traits/WeightingPower.hpp#L30-L49)
```
    /** Describe if a particle attribute describes the quantity of a macro
     *  particle
     *
     * Depending on the implementation of an attribute, it might be sometimes
     * useful to return a quantity regarding one of the underlying real
     * particles (\see MacroWeighted).
     *
     * This trait defines how each attribute needs to be scaled with the
     * weighting (linear, quadratic, ...) to convert between "real" and "macro"
     * particle attributes.
     *   \see http://www.openPMD.org
     *   \see http://dx.doi.org/10.5281/zenodo.33624
     *   \see https://git.io/vwlWa
     *
     * \tparam T_Identifier any picongpu identifier
     * \return \p float_64 ::get() as static public method
     *
     */
```
  - [`MacroWeighted`](https://github.com/ComputationalRadiationPhysics/picongpu/blob/71cfb03ec813c88a7e6f304b870f8731093b3b4f/src/picongpu/include/particles/traits/MacroWeighted.hpp#L30-L52)
```
    /** Describe if a particle attribute describes the quantity of a macro
     *  particle
     *
     * Depending on the implementation of an attribute, it might be sometimes
     * useful to return a quantity regarding one of the underlying real
     * particles (false: "this attribute is not weighted accordingly for the
     * whole ensemble of particles in its macro particle) or just handle the
     * whole macro particle at once
     * (true: "this attribute is already weighted").
     *
     * This trait defines for each attribute if it needs to be scaled with the
     * weighting. *How* the scaling with weighting is applied can be seen in
     * \see WeightingPower
     *   \see http://www.openPMD.org
     *   \see http://dx.doi.org/10.5281/zenodo.33624
     *   \see https://git.io/vwlWa
     *
     * \tparam T_Identifier any picongpu identifier
     * \return \p bool ::get() as static public method
     *
     */
```

**Comment by ax3l** [comment]
I will add some inline-comments to those files in `speciesAttributes.unitless/param` for now

**Comment by ax3l** [comment]
Added inline docs in #1483.

The *hierarchy* of param and their corresponding unitless files is then a task documented in #12, so feel free to close this one if your questions are answered :)

**Comment by n01r** [comment]
Thx @ax3l! :smile:  This came up when @PrometheusPi wanted to introduce a new attribute to have it available later during analysis. Even though these are developer tools I think it wouldn't hurt to at least mention in the wiki which files have to be touched when adding attributes. 

I will close the issue once @PrometheusPi is finished with reviewing your PR and it has been merged.

**Comment by ax3l** [comment]
Yes, from our definition users should use pre-defined attributes.
Power-Users and developers want to add new ones, the user-doc should then fluently forward to the next page in developers giving all the programming details an average user does not need to know.

**Comment by n01r** [comment]
There is now a link in the Wiki from *Particle attributes* in *User Documentation* to *Create a new attribute* in *Developer Documentation*. I think the issue can be closed.

**Comment by ax3l** [comment]
thanks. this needs migration to sphinx later on, so I change the scope and leave it open

**Comment by ax3l** [comment]
Needs to be added to the user workflows section: http://picongpu.readthedocs.io/en/dev/usage/workflows.html
